multimodal/src/autogluon/multimodal/optimization/lit_mmdet.py

```diff
@@ -0,0 +1,317 @@
+import logging
+from typing import Callable, Dict, List, Optional, Union
+
+import numpy as np
+import pytorch_lightning as pl
+import torch
+import torchmetrics
+from omegaconf import DictConfig
+from torch import nn
+from torch.nn.modules.loss import _Loss
+from torchmetrics.aggregation import BaseAggregator
+
+try:
+    import mmdet
+    from mmcv import ConfigDict
+except ImportError:
+    pass
+
+from ..constants import AUTOMM
+from ..utils import unpack_datacontainers
+from .utils import apply_layerwise_lr_decay, apply_single_lr, apply_two_stages_lr, get_lr_scheduler, get_optimizer
+
+logger = logging.getLogger(AUTOMM)
+
+
+class MMDetLitModule(pl.LightningModule):
+    def __init__(
+        self,
+        model,
+        optim_type: Optional[str] = None,
+        lr_choice: Optional[str] = None,
+        lr_schedule: Optional[str] = None,
+        lr: Optional[float] = None,
+        lr_decay: Optional[float] = None,
+        end_lr: Optional[Union[float, int]] = None,
+        lr_mult: Optional[Union[float, int]] = None,
+        weight_decay: Optional[float] = None,
+        warmup_steps: Optional[int] = None,
+        loss_func: Optional[_Loss] = None,
+        validation_metric: Optional[torchmetrics.Metric] = None,
+        validation_metric_name: Optional[str] = None,
+        custom_metric_func: Callable = None,
+        test_metric: Optional[torchmetrics.Metric] = None,
+        efficient_finetune: Optional[str] = None,
+    ):
+        super().__init__()  # TODO: inherit LitModule
+        self.save_hyperparameters(
+            ignore=[
+                "model",
+                "validation_metric",
+                "test_metric",
+                "loss_func",
+                "matches",
+            ]
+        )
+        self.model = model
+        self.validation_metric = validation_metric
+        self.validation_metric_name = f"val_{validation_metric_name}"
+        if isinstance(validation_metric, BaseAggregator) and custom_metric_func is None:
+            raise ValueError(
+                f"validation_metric {validation_metric} is an aggregation metric,"
+                f"which must be used with a customized metric function."
+            )
+        self.custom_metric_func = custom_metric_func
+        self.id2label = dict(zip(range(100), range(100)))  # TODO: replace with real id2label
```

Comment:
why 100?

Reply:
Replaced this hardcode with a id2label in self.model.